pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/DSMServer.java

```diff
@@ -543,70 +540,83 @@ protected void setupCustomRouting(@NonNull Config cfg) {
+                cfg.hasPath(ApplicationConfigConstants.DDP_ENCODED) && cfg.getBoolean(ApplicationConfigConstants.DDP_ENCODED);
         String auth0Domain = cfg.getString(ApplicationConfigConstants.AUTH0_DOMAIN);
         String auth0claimNameSpace = cfg.getString(ApplicationConfigConstants.AUTH0_CLAIM_NAMESPACE);
 
         if (StringUtils.isBlank(bspSecret)) {
-            throw new RuntimeException("No secret supplied for BSP endpoint, system exiting.");
+            throw new DsmInternalError("No secret supplied for BSP endpoint, system exiting.");
         }
 
         if (StringUtils.isBlank(ddpSecret)) {
-            throw new RuntimeException("No secret supplied for DDP endpoint, system exiting.");
+            throw new DsmInternalError("No secret supplied for DDP endpoint, system exiting.");
         }
 
         String appRoute = cfg.hasPath("portal.appRoute") ? cfg.getString("portal.appRoute") : null;
 
         if (StringUtils.isBlank(appRoute)) {
-            throw new RuntimeException("appRoute was not configured correctly.");
+            throw new DsmInternalError("appRoute was not configured correctly.");
         }
 
-        //  capture basic route info for logging
-        //        before("*", new LoggingFilter(auth0Domain, auth0claimNameSpace));
-        before(apiRoot + "*", new LoggingFilter(auth0Domain, auth0claimNameSpace, bspSecret, null, bspSecretEncoded));
-        before(uiRoot + "*", new LoggingFilter(auth0Domain, auth0claimNameSpace, null, null, false));
-        before(infoRoot + "*", new LoggingFilter(auth0Domain, auth0claimNameSpace, ddpSecret, signer, ddpSecretEncoded));
-        before(appRoute + "*", new LoggingFilter(auth0Domain, auth0claimNameSpace, ddpSecret, signer, ddpSecretEncoded));
+        before(API_ROOT + "*", new LoggingFilter(auth0Domain, auth0claimNameSpace, bspSecret, null, bspSecretEncoded));
+        before(DSM_ROOT + "*",
+                new LoggingFilter(auth0Domain, auth0claimNameSpace, dsmSecretForJuniper, juniperSigner, juniperSecretEncoded));
+        before(UI_ROOT + "*", new LoggingFilter(auth0Domain, auth0claimNameSpace, null, null, false));
+        before(INFO_ROOT + "*", new LoggingFilter(auth0Domain, auth0claimNameSpace, ddpSecret, KDUX_SIGNER, ddpSecretEncoded));
+        before(appRoute + "*", new LoggingFilter(auth0Domain, auth0claimNameSpace, ddpSecret, KDUX_SIGNER, ddpSecretEncoded));
         afterAfter((req, res) -> MDC.clear());
 
-        before(apiRoot + "*", (req, res) -> {
+        before(API_ROOT + "*", (req, res) -> {
             if (!new JWTRouteFilter(auth0Domain).isAccessAllowed(req, false, bspSecret)) {
                 logger.info("Returning 401 because token was not verified");
                 halt(401);
             }
             res.header(HttpHeaders.CONTENT_TYPE, MediaType.JSON_UTF_8.toString());
         });
 
+        before(DSM_ROOT + "*", (req, res) -> {
```

Comment:
It looks like here and above that the generic DSM_ROOT is dedicated to Juniper.
IOW, will all users of this path be Juniper?
If so, does it make sense to make this more generic now?

Reply:
I _think_ given the current predicted lifetime for DSM (that I know of) this route will be only called by Juniper to create kits in Pepper DSM.
However, the api is designed to work for any non-DSS kit creation. Given that, I'd say if time comes for other users to use this api to trigger a kit creation, we would need to make changes in our authentication methods to make sure an endpoint can work with more than one possible secret (to avoid sharing Juniper's secret with other users)

Since Juniper's secret should definitely be different than GP's and the endpoint should go into a different category than the one GP uses, I thought this much change would be enough for the scope of this work given we don't know how long DSM will live, or even if other future Juniper studies will use Pepper DSM or not. 

What do you think?

Reply:
Yup, that makes sense.
I think a comment on why we are handling a generic path (dsm/) with a Juniper specific handler would be helpful. It could be as simple as "This path only handles Juniper for now."